xclim/run_length.py

```diff
@@ -83,3 +84,86 @@ def longest_run(arr):
     """
     v, rl = rle(arr)[:2]
     return np.where(v, rl, 0).max()
+
+
+def xr_longest_run(da, freq):
+    """Return the length of the longest run of true values along the time dimension.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+      Boolean array with time coordinates.
+    freq : str
+      Resampling frequency.
+
+    """
+    # Create an monotonously increasing index [0,1,2,...] along the time dimension.
+    i = xr.DataArray(np.arange(da.time.size), dims='time')
+    index = xr.broadcast(i, da)[0]
+
+    ini = xr.DataArray([-1], coords={'time': da.time[:1]}, dims='time')
+    end = xr.DataArray([da.time.size], coords={'time': da.time[-1:]}, dims='time')
+
+    masked_da = xr.concat((ini, index.where(~da), end), dim='time')
+
+    # Fill NaNs with the following valid value
+    nan_filled_da = masked_da.bfill(dim='time')
+
+    # Find the difference between start and end indices
+    diff_ind = nan_filled_da.diff(dim='time') - 1
+
+    # Find the longest run by period - but it does not work if all values are True.
+    run = diff_ind.resample(time=freq).max(dim='time')
+    return run
+    g = da.resample(time=freq)
+    return run.where(~g.all(dim='time'), g.count(dim='time'))
```

Comment:
Two returns?

Reply:
Oups. This function is going to the garbage after the benchmark.